lib/school_house_web/views/html_helpers.ex

```diff
@@ -9,9 +9,8 @@ defmodule SchoolHouseWeb.HtmlHelpers do
 
   def avatar_url(github_link), do: "#{github_link}.png"
 
-  def current_page_locale_link(conn, locale) do
-    current_url = Phoenix.Controller.current_url(conn)
-    String.replace(current_url, current_locale(), locale)
+  def current_page_locale_path(%{request_path: request_path}, locale) do
```

Comment:
I'm open to either or on this but I'd love to get your feedback @jaimeiniesta:

Do you have any strong feelings about matches like this including the struct or are maps fine? Folks seem to prefer one or the other and before we get too far I wanted to address that.

Expressed in code:

```elixir
def current_page_locale_path(%Plug.Conn{request_path: request_path}, locale) do

def current_page_locale_path(%{request_path: request_path}, locale) do
````

Reply:
I prefer the maps most of the time, as this is internal code and we are only increasing complexity / adding dependency on another module without reason. Map matching should be enough.

Reply:
Thanks for chiming in @gemantzu :tada: 

That is kind of my thinking too. On one hand using the struct makes the code's intention clearer. On the other though, it can really start to clutter up code and it adds the tiniest bit of extra work to testing these functions directly.

Reply:
yeah, this can come to bite us, even though I don't see this project being that big to be too much of an issue tbh.

Reply:
Yes I also think that maps are good enough for this. We just care about the `:request_path` key being present, not so much about the exact module.

Reply:
Thank you for the input @jaimeiniesta and @gemantzu. It shall be so :tada: